librz/core/cbin.c

```diff
@@ -790,16 +790,14 @@ RZ_API bool rz_core_bin_apply_maps(RzCore *core, RzBinFile *binfile, bool va) {
 
 	RzListIter *it;
 	RzBinMap *map;
-	// reverse because of the way io handles maps
```

Comment:
Do you know what did this mean? I guess there was a reason to handling things in reverse order... is that not relevant anymore? Maybe adding a bit of explanation of why this is not relevant anymore would help even for future reference.

Reply:
@ret2libc In the past RzIO would go through the maps array backwards to construct the skyline vector. This isn't true anymore

Reply:
wasn't the reason for that to assign the proper priorities to each maps? I have fuzzy memories about that.

Reply:
I think @thestr4ng3r knows best of this part's changes.

Reply:
This is my last knowledge of the priorities:

https://github.com/rizinorg/rizin/blob/5f5c0d66dfdc01db1af782bb024c2873fdd27a1a/librz/include/rz_io.h#L73

Which I think makes sense, to specify the priorities by their position in the vector. Whether it's reverse or not is then merely convention (and maybe slightly relevant for insertion/removal performance).

> This isn't true anymore

Was this changed at some point?

Reply:
@thestr4ng3r I checked again and confirmed that new maps are pushed to the end of the vector as they are added, and they have higher priority, this never changed, what changed was an implementation detail of how the skyline is created and that should be unrelated to this.
Why this code bothered to go through maps in reverse order, I don't know.
This would only work if it was a requirement that `RzBinObject->maps` always had the highest priority maps as the first elements, but there isn't anywhere where that that is specified, so this never was a guarantee anyway.

Reply:
Good. Lets see if @thestr4ng3r approves and lets merge this. 👍